library/src/main/java/com/novoda/downloadmanager/lib/DownloadsRepository.java

```diff
@@ -65,15 +66,20 @@ public DownloadsRepository(SystemFacade systemFacade, ContentResolver contentRes
         return getAllDownloadsFor(NO_BATCH_ID);
     }
 
+    @Nullable
     public FileDownloadInfo getDownloadFor(long id) {
         Uri uri = ContentUris.withAppendedId(downloadsUriProvider.getAllDownloadsUri(), id);
         Cursor downloadsCursor = contentResolver.query(uri, null, null, null, null);
-        try {
-            downloadsCursor.moveToFirst();
-            FileDownloadInfo.Reader reader = new FileDownloadInfo.Reader(contentResolver, downloadsCursor);
-            return downloadInfoCreator.create(reader);
-        } finally {
-            downloadsCursor.close();
+        if (downloadsCursor != null) {
```

Comment:
I think it would probably be clearer to have a quick return if null here - so something like
```
if (downloadsCursor != null) {
    return null;
```
Makes it a bit easier to see exactly what gets returned if the cursor is null and removes a level of indentation too.